test/JDBC/expected/BABEL_4302.out

```diff
@@ -5823,3 +5823,357 @@ GO
 
 ~~ERROR (Message: Arithmetic overflow error converting expression to data type datetime.)~~
 
+
+
+
+-- Max Limits checked for datatypes
+SELECT DATEPART(day, CAST(9999999999 AS bit));
+GO
+~~START~~
+int
+2
+~~END~~
+
+
+SELECT DATEPART(week, CAST(922337203685477.5807 AS money));
+GO
+~~START~~
+int
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Arithmetic overflow error converting expression to data type datetime.)~~
+
+
+SELECT DATEPART(weekday, CAST(214748.3649 AS smallmoney));
+GO
+~~START~~
+int
+~~ERROR (Code: 220)~~
+
+~~ERROR (Message: value for domain smallmoney violates check constraint "smallmoney_check")~~
+
+
+SELECT DATEPART(year, CAST(2147483647 AS int));
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Arithmetic overflow error converting expression to data type datetime.)~~
+
+
+SELECT DATEPART(q, CAST(32767 AS smallint));
+GO
+~~START~~
+int
+3
+~~END~~
+
+
+SELECT DATEPART(qq, CAST(9223372036854775807 AS bigint));
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Arithmetic overflow error converting expression to data type datetime.)~~
+
+
+SELECT DATEPART(second, CAST(256 AS tinyint));
+GO
+~~START~~
+int
+~~ERROR (Code: 220)~~
+
+~~ERROR (Message: value for domain tinyint violates check constraint "tinyint_check")~~
```

Comment:
Is this the same T-SQL behaviour? I dont think so. Seems like some bug with casting 256 to TINYINT. Please confirm and file a bug if needed

Reply:
seems like every domain data type has this problem, for example, smallmoney

Reply:
This is a issue of cast function. For the datatypes, the overflow error message is thrown from the cast function before hitting the datepart functions. Will fiile a bug